helix-core/src/main/java/org/apache/helix/task/TaskUtil.java

```diff
@@ -220,6 +221,85 @@ protected static void setWorkflowContext(HelixManager manager, String workflowRe
   }
 
   /**
+   * Intialize the user content store znode setup
+   * @param manager          a connection to Helix
+   * @param workflowResource the name of workflow
+   * @param jobResource      the name of job
+   * @param record           the initial data
+   */
+  protected static void setUserContent(HelixManager manager, String workflowResource,
+      String jobResource, ZNRecord record) {
+    manager.getHelixPropertyStore().create(Joiner.on("/")
```

Comment:
This creates both workflow content node and job content node everytime, which is not needed.  Since the workflow and job directory structure in propertyStore are same /WorkflowOrJobName/usercontent, this method can be generic for both workflow and job, such as
SetUserContent(HelixManager manager,  String workflowJobResource, ...):
           manager.getHelixPropertyStore().create(Joiner.on("/")            .join(TaskConstants.REBALANCER_CONTEXT_ROOT, workflowJobResource, TaskUtil.USER_CONTENT_NODE),
       record, AccessOption.PERSISTENT);